python_modules/dagster-graphql/dagster_graphql_tests/graphql/repo_definitions.py

```diff
@@ -0,0 +1,43 @@
+from contextlib import contextmanager
+
+from dagster import (
+    DagsterInstance,
+    Definitions,
+    _check as check,
+    asset,
+)
+from dagster._config.structured_config import Resource
+from dagster._core.definitions.repository_definition import SINGLETON_REPOSITORY_NAME
+from dagster_graphql.test.utils import define_out_of_process_context
+
+
+@asset
+def my_asset():
+    pass
+
+
+class MyResource(Resource):
+    """my description"""
+
+    a_string: str = "baz"
+    a_bool: bool
+    an_unset_string: str = "defaulted"
+
+
+__repository__ = Definitions(
```

Comment:
why is this variable named `__repository__`

Reply:
This is a good question. For some reason, the execution path in `test_top_level_resources.py` doesn't pick up on this `Definitions` object and its associated repo unless it's called `__repository__`. 

The output of the above (now removed 😅 ) print statements, using `defs = ...`:

```python
[]
[SerializableErrorInfo(message='dagster._core.errors.DagsterInvariantViolationError: __repository__ not found at module scope in file /Users/ben/Documents/repos/dagster/python_modules/dagster-graphql/dagster_graphql_tests/graphql/repo_definitions.py.\n', stack=['  File "/Users/ben/Documents/repos/dagster/python_modules/dagster/dagster/_grpc/server.py", line 256, in __init__\n    self._loaded_repositories: Optional[LoadedRepositories] = LoadedRepositories(\n', '  File "/Users/ben/Documents/repos/dagster/python_modules/dagster/dagster/_grpc/server.py", line 114, in __init__\n    loadable_targets = get_loadable_targets(\n', '  File "/Users/ben/Documents/repos/dagster/python_modules/dagster/dagster/_grpc/utils.py", line 33, in get_loadable_targets\n    attribute, load_def_in_python_file(python_file, attribute, working_directory)\n', '  File "/Users/ben/Documents/repos/dagster/python_modules/dagster/dagster/_core/definitions/reconstruct.py", line 666, in load_def_in_python_file\n    return def_from_pointer(CodePointer.from_python_file(python_file, attribute, working_directory))\n', '  File "/Users/ben/Documents/repos/dagster/python_modules/dagster/dagster/_core/definitions/reconstruct.py", line 672, in def_from_pointer\n    target = pointer.load_target()\n', '  File "/Users/ben/Documents/repos/dagster/python_modules/dagster/dagster/_core/code_pointer.py", line 179, in load_target\n    return _load_target_from_module(\n', '  File "/Users/ben/Documents/repos/dagster/python_modules/dagster/dagster/_core/code_pointer.py", line 202, in _load_target_from_module\n    raise DagsterInvariantViolationError(f"{fn_name} not found {error_suffix}")\n'], cls_name='DagsterInvariantViolationError', cause=None, context=None)]
```

Reply:
updated, I think I had plugged in `SINGLETON_REPOSITORY_NAME` for the name of the module-level object to load into the test instance